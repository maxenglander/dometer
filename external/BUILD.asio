filegroup(
    name = "asio_headers",
    srcs = glob([
        "include/asio.hpp",
        "include/asio/**/*.hpp",
        "include/asio/**/*.ipp",
    ]),
)

genrule(
    name = "asio_sources",
    srcs = [],
    outs = ["asio.cpp"],
    cmd = "echo '#include \"asio/impl/src.hpp\"' > $@",
)

cc_library(
    name = "asio",
    data = [
        ":asio_headers",
        ":asio_sources",
    ],
    copts = [
        "-DASIO_DYN_LINK=1",
        "-DASIO_STANDALONE=1",
        "-Iexternal/net_sourceforge_asio_asio/include",
    ],
    hdrs = [
        ":asio_headers",
    ],
    srcs = [
        ":asio_sources",
    ],
    linkopts = [
        "-lpthread",
    ],
    strip_include_prefix = "include",
    visibility = ["//visibility:public"],
)


