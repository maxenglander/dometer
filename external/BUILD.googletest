cc_library(
    name = "gtest",
    srcs = [
        "googlemock/src/gmock-all.cc",
        "googletest/src/gtest-all.cc",
    ],
    hdrs = glob([
        "**/*.h",
        "googlemock/src/*.cc",
        "googletest/src/*.cc",
    ]),
    includes = [
        "googlemock",
        "googletest",
        "googlemock/include",
        "googletest/include",
    ],
    linkopts = ["-lpthread"],
    visibility = ["//visibility:public"],
)

cc_library(
    name = "gtest_main",
    srcs = [
        "googlemock/src/gmock_main.cc",
    ],
    linkopts = ["-lpthread"],
    visibility = ["//visibility:public"],
    deps = [":gtest"],
)
