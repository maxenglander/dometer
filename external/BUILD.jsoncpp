package(
    default_visibility = ["//visibility:public"],
)

cc_library(
    name = "jsoncpp",
    srcs = glob([
        "src/lib_json/*.cpp",
        "src/lib_json/*.h",
    ]),
    hdrs = glob([
        "include/json/*.h",
    ]),
    copts = [
        "-DJSON_USE_EXCEPTION=0",
        "-Iexternal/jsoncpp/include",
        "-Iexternal/jsoncpp/include/jsoncpp",
    ],
    strip_include_prefix = "include",
    visibility = ["//visibility:public"],
    deps = [":private"],
)

cc_library(
    name = "private",
    textual_hdrs = ["src/lib_json/json_valueiterator.inl"],
)
