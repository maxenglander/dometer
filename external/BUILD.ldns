load("@rules_foreign_cc//tools/build_defs:configure.bzl", "configure_make")

package(
    default_visibility = ["//visibility:public"],
)

filegroup(
    name = "ldns_all",
    srcs = glob(["**"])
)

configure_make(
    name = "ldns_configure_make",
    configure_in_place = True,
    configure_options = [
        "--disable-dane",
        "--disable-ecdsa",
        "--disable-gost",
        "--disable-sha2",
        "--without-ssl"
    ],
    lib_source = ":ldns_all",
    out_lib_dir = "lib",
    shared_libraries = ["libldns.so"],
    visibility = ["//visibility:public"],
)
