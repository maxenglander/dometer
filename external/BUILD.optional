cc_library(
    name = "optional",
    hdrs = [
        "include/tl/optional.hpp",
    ],
    includes = ["include"],
    visibility = ["//visibility:public"],
)
