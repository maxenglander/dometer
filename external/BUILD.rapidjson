cc_library(
    name = "rapidjson",
    hdrs = glob([
        "include/rapidjson/error/*.h",
        "include/rapidjson/internal/*.h",
        "include/rapidjson/msinttypes/*.h",
        "include/rapidjson/*.h",
    ]),
    includes = ["include"],
    visibility = ["//visibility:public"],
)
