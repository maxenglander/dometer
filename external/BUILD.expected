cc_library(
    name = "expected",
    hdrs = [
        "include/tl/expected.hpp",
    ],
    includes = ["include"],
    visibility = ["//visibility:public"],
)
